### PR TITLE
Add support for allowing subusers to access SFTP.

### PR DESCRIPTION
**Pull request by ShadowKitten**
- Fixed unit tests.
- Modified some permissions for more friendly naming.
- Removed subuser SFTP permissions in favor of single 'SFTP Allowed' permission.

### CODE REVIEW

**Comment by ShadowKitten** [comment]
I went ahead and reverted the permission changes as there is currently no way to update the daemon with the new permissions mappings. One permission mapping s:files:download is still going to be needed to be updated on the daemon for each subuser so they will be able to download files from SFTP.

I haven't been able to find a good solution to this yet besides possibly adding a CLI for creating and updating subusers.

File: app/Models/Permission.php [hunk 1]
```diff
@@ -90,12 +90,12 @@ class Permission extends Model implements CleansAttributes, ValidableContract
...
-        'sftp' => [
-            'view-sftp' => null,
-            'view-sftp-password' => null,
-            'reset-sftp' => 's:set-password',
+        'database' => [
+            'view-databases' => null,
+            'reset-db-password' => null,
         ],
         'file' => [
+            'view-sftp' => null,
```
Comment (DaneEveritt) [inline app/Models/Permission.php]: This should be renamed to `access-sftp` for clarity I think.

File: app/Services/Sftp/AuthenticateUsingPasswordService.php [hunk 2]
```diff
@@ -25,21 +26,29 @@ class AuthenticateUsingPasswordService
...
+     * @param \Pterodactyl\Contracts\Repository\ServerRepositoryInterface  $repository
+     * @param \Pterodactyl\Contracts\Repository\UserRepositoryInterface    $userRepository
+     * @param \Pterodactyl\Contracts\Repository\SubuserRepositoryInterface $subuserRepository
      */
     public function __construct(
         DaemonKeyProviderService $keyProviderService,
         ServerRepositoryInterface $repository,
-        UserRepositoryInterface $userRepository
+        UserRepositoryInterface $userRepository,
+        SubuserRepositoryInterface $subuserRepository
```
Comment (DaneEveritt) [inline app/Services/Sftp/AuthenticateUsingPasswordService.php]: Please keep these in alphabetical order, same for the variable setting in the constructor itself (and the order at the top of the file).

File: app/Services/Sftp/AuthenticateUsingPasswordService.php [hunk 3]
```diff
@@ -73,10 +82,19 @@ public function handle(string $username, string $password, int $node, string $se
         }
 
         $server = $this->repository->setColumns(['id', 'node_id', 'owner_id', 'uuid', 'installed', 'suspended'])->getByUuid($server);
-        if ($server->node_id !== $node || (! $user->root_admin && $server->owner_id !== $user->id)) {
+        if ($server->node_id !== $node) {
             throw new RecordNotFoundException;
         }
 
+        if ((! $user->root_admin && $server->owner_id !== $user->id)) {
```
Comment (DaneEveritt) [inline app/Services/Sftp/AuthenticateUsingPasswordService.php]: The double parenthesis are not needed here.

File: tests/Unit/Services/Sftp/AuthenticateUsingPasswordServiceTest.php [hunk 4]
```diff
@@ -39,6 +45,7 @@ public function setUp()
         $this->keyProviderService = m::mock(DaemonKeyProviderService::class);
         $this->repository = m::mock(ServerRepositoryInterface::class);
         $this->userRepository = m::mock(UserRepositoryInterface::class);
+        $this->subuserRepository = m::mock(SubuserRepositoryInterface::class);
```
Comment (DaneEveritt) [inline tests/Unit/Services/Sftp/AuthenticateUsingPasswordServiceTest.php]: Same as above with the alphabetical order here.

**Pull request merged**